packages/connectivity_plus/connectivity_plus/android/src/main/java/dev/fluttercommunity/plus/connectivity/Connectivity.java

```diff
@@ -23,58 +25,75 @@ public Connectivity(ConnectivityManager connectivityManager) {
     if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
       Network network = connectivityManager.getActiveNetwork();
       NetworkCapabilities capabilities = connectivityManager.getNetworkCapabilities(network);
       if (capabilities == null) {
-        return CONNECTIVITY_NONE;
+        types.add(CONNECTIVITY_NONE);
+        return types;
       }
       if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
-        return CONNECTIVITY_WIFI;
+        types.add(CONNECTIVITY_WIFI);
       }
       if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
-        return CONNECTIVITY_ETHERNET;
+        types.add(CONNECTIVITY_ETHERNET);
       }
       if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_VPN)) {
-        return CONNECTIVITY_VPN;
+        types.add(CONNECTIVITY_VPN);
       }
       if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
-        return CONNECTIVITY_MOBILE;
+        types.add(CONNECTIVITY_MOBILE);
       }
       if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_BLUETOOTH)) {
-        return CONNECTIVITY_BLUETOOTH;
+        types.add(CONNECTIVITY_BLUETOOTH);
       }
+      if (types.isEmpty()) {
+        types.add(CONNECTIVITY_NONE);
+        return types;
+      }
+    } else {
+      // For legacy versions, return a single type as before or adapt similarly if multiple types need to be supported
+      return getNetworkTypesLegacy();
     }
 
-    return getNetworkTypeLegacy();
+    return types;
   }
 
   @SuppressWarnings("deprecation")
-  private String getNetworkTypeLegacy() {
+  private List<String> getNetworkTypesLegacy() {
     // handle type for Android versions less than Android 6
     android.net.NetworkInfo info = connectivityManager.getActiveNetworkInfo();
+    List<String> types = new ArrayList<>();
     if (info == null || !info.isConnected()) {
-      return CONNECTIVITY_NONE;
+      types.add(CONNECTIVITY_NONE);
+      return types;
     }
     int type = info.getType();
     switch (type) {
       case ConnectivityManager.TYPE_BLUETOOTH:
-        return CONNECTIVITY_BLUETOOTH;
+        types.add(CONNECTIVITY_BLUETOOTH);
+        break;
       case ConnectivityManager.TYPE_ETHERNET:
-        return CONNECTIVITY_ETHERNET;
+        types.add(CONNECTIVITY_ETHERNET);
+        break;
       case ConnectivityManager.TYPE_WIFI:
       case ConnectivityManager.TYPE_WIMAX:
-        return CONNECTIVITY_WIFI;
+        types.add(CONNECTIVITY_WIFI);
+        break;
       case ConnectivityManager.TYPE_VPN:
-        return CONNECTIVITY_VPN;
+        types.add(CONNECTIVITY_VPN);
+        break;
       case ConnectivityManager.TYPE_MOBILE:
       case ConnectivityManager.TYPE_MOBILE_DUN:
       case ConnectivityManager.TYPE_MOBILE_HIPRI:
-        return CONNECTIVITY_MOBILE;
+        types.add(CONNECTIVITY_MOBILE);
+        break;
       default:
-        return CONNECTIVITY_NONE;
+        types.add(CONNECTIVITY_NONE);
```

Comment:
Would it be possible to simplify this by returning a `List.of(type)`? For what I understand, the array will always have just one item.

I haven't done Java in a long time, so I don't remember if that was possible :)

Reply:
I thinks it's not so necessary cause return executed straight after switch statement.

Reply:
what I mean is: 
- the code is creating an empty array on line 74
- then adding a single item in the switch statement
- then returning the array at the end after the switch.

And I'd prefer if the code could just call return inside the switch case with a `List.of(item)`. It is less code, and less prone to returning an empty array in case of a future code refactor.

Reply:
@miquelbeltran Thanks for the suggestion! I aimed to optimize by avoiding extra array creation, but I'm open to your perspective. If you think changing it is best, I'm on board.